File: tasks/dfsConnectedComponents.java
int dfsConnectedComponents(final boolean[][] matrix) {

  class Helper {
  void dfs(int currentVertex, boolean[] visited) {
    visited[currentVertex] = true;
    for (int nextVertex = 0; nextVertex < matrix[0].length; nextVertex++) {
      if (!matrix[currentVertex][nextVertex] || visited[nextVertex])
        continue;
      dfs(nextVertex, visited);
    }
  }
  }
  Helper h = new Helper();

  boolean[] visited = new boolean[matrix.length];
  int componentsCount = 0;

  for (int startVertex = 0; startVertex < matrix.length; startVertex++) {
    if (!visited[startVertex]) {
      h.dfs(startVertex, visited);
      componentsCount++;
    }
  }

  return componentsCount;
}

File: tasks/dfsConnectedComponents.py
def dfsConnectedComponents(matrix):

    def dfs(currentVertex, visited):
        visited[currentVertex] = True
        for next_vertex in range(len(matrix[0])):
          if not matrix[currentVertex][next_vertex] or visited[next_vertex]:
            continue
          dfs(next_vertex, visited)

    visited = []
    componentsCount = 0

    for i in range(len(matrix)):
        visited.append(False)

    for startVertex in range(len(matrix)):
        if not visited[startVertex]:
            dfs(startVertex, visited)
            componentsCount += 1

    return componentsCount
